projects/driver_controls/src/power_fsm.c

```diff
@@ -7,11 +7,22 @@
 // Power FSM state definitions
 
 FSM_DECLARE_STATE(state_off);
+FSM_DECLARE_STATE(state_off_brake);
+FSM_DECLARE_STATE(state_charging);
 FSM_DECLARE_STATE(state_on);
 
 // Power FSM transition table definitions
 
 FSM_STATE_TRANSITION(state_off) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_POWER, state_charging);
+  FSM_ADD_TRANSITION(INPUT_EVENT_MECHANICAL_BRAKE_PRESSED, state_off_brake);
+}
+
+FSM_STATE_TRANSITION(state_off_brake) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_POWER, state_on);
+}
+
+FSM_STATE_TRANSITION(state_charging) {
```

Comment:
Why does toggling power twice put it into "on"? That seems a little counterintuitive. @ckitagawa @taipingli ?

Reply:
That's how most cars with start buttons work. If you hit the start button without pressing the break it just boots the car's electrical systems without turning on the engine. Doing it again with the pedal pressed will start the engine otherwise if the pedal is not pressed the car turns off again.